### PR TITLE
zoraxy: init at 3.0.5

### LINKED ISSUE #312814: Package request: Zoraxy
**Project description**

"A general purpose HTTP reverse proxy and forwarding tool."

**Metadata**

* homepage URL: https://zoraxy.arozos.com/
* source URL: https://github.com/tobychui/zoraxy
* license: agpl-3.0
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #312814

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#zoraxy
> tree result
 result
└──  bin
   └──  zoraxy
```

Tested `./result/bin/zoraxy` creating a new account

File: pkgs/by-name/zo/zoraxy/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  ...
```
Comment (SuperSandro2000) [inline pkgs/by-name/zo/zoraxy/package.nix:5]:
```suggestion
  fetchFromGitHub
```

File: pkgs/by-name/zo/zoraxy/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  sourceRoot = "${src.name}/src";
+
+  vendorHash = "sha256-YI6LSccPDnVhGyPIEFIF41ex0WJlHtb3nP+8+1G/LA0=";
+
+  # For some reason the acme_test.go references a stackoverflow.pem file not present in the repository
+  doCheck = false;
+
+  meta = {
+    description = "A general purpose HTTP reverse proxy and forwarding tool written in Go.";
```
Comment (SuperSandro2000) [inline pkgs/by-name/zo/zoraxy/package.nix:25]:
```suggestion
    description = "A general purpose HTTP reverse proxy and forwarding tool written in Go";
```

File: pkgs/by-name/zo/zoraxy/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    repo = "zoraxy";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-tBEIePauoZHs7aaVWCr9l8nvMT9YUDuUK9oTh7iwSRw=";
+  };
+
+  sourceRoot = "${src.name}/src";
+
+  vendorHash = "sha256-YI6LSccPDnVhGyPIEFIF41ex0WJlHtb3nP+8+1G/LA0=";
+
+  # For some reason the acme_test.go references a stackoverflow.pem file not present in the repository
```
Comment (SuperSandro2000) [inline pkgs/by-name/zo/zoraxy/package.nix:21]: Can we only ignore that one test?

File: pkgs/by-name/zo/zoraxy/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  ...
+}:
+buildGoModule rec {
+  pname = "zoraxy";
+  version = "3.0.4";
```
Comment (SuperSandro2000) [inline pkgs/by-name/zo/zoraxy/package.nix:9]:
```suggestion
}:

buildGoModule rec {
  pname = "zoraxy";
  version = "3.0.4";

```

File: pkgs/by-name/zo/zoraxy/package.nix (hunk 3 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/zo/zoraxy/package.nix:21]: i don't really know how to ignore a single test. Do you have some docs where I can look up?

**Comment by luftmensch-luftmensch** [comment]
@SuperSandro2000 Applied suggestion.

Note: For the tests I had a few, but should be good to go now

**Comment by luftmensch-luftmensch** [comment]
@SuperSandro2000 bumped version to 3.0.5

[Release notes](https://github.com/tobychui/zoraxy/releases/tag/3.0.5)

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**